specs/eip4844/p2p-interface.md

```diff
@@ -139,45 +155,56 @@ Per `context = compute_fork_digest(fork_version, genesis_validators_root)`:
 | `CAPELLA_FORK_VERSION`   | `capella.SignedBeaconBlock`   |
 
-#### BeaconBlockAndBlobsSidecarByRoot v1
+No more than `MAX_REQUEST_BLOCKS_EIP4844` may be requested at a time.
 
-**Protocol ID:** `/eth2/beacon_chain/req/beacon_block_and_blobs_sidecar_by_root/1/`
+#### BlobSidecarsByRoot v1
+
+**Protocol ID:** `/eth2/beacon_chain/req/blob_sidecars_by_root/1/`
 
 Request Content:
 
+```python
+class BlobIdentifier(Container):
+    block_root: Root
+    index: uint64
+```
+
 ```
 (
-  List[Root, MAX_REQUEST_BLOCKS]
+  List[BlobIdentifier, MAX_REQUEST_BLOCKS_EIP4844]
 )
 ```
 
 Response Content:
 
 ```
 (
-  List[SignedBeaconBlockAndBlobsSidecar, MAX_REQUEST_BLOCKS]
+  List[BlobSidecar, MAX_REQUEST_BLOCKS_EIP4844]
 )
 ```
 
-Requests blocks by block root (= `hash_tree_root(SignedBeaconBlockAndBlobsSidecar.beacon_block.message)`).
-The response is a list of `SignedBeaconBlockAndBlobsSidecar` whose length is less than or equal to the number of requests.
+Requests sidecars by block root and index.
+The response is a list of `BlobSidecar` whose length is less than or equal to the number of requests.
 It may be less in the case that the responding peer is missing blocks and sidecars.
 
-No more than `MAX_REQUEST_BLOCKS` may be requested at a time.
+The response is unsigned, i.e. `BlobSidecar`, as the signature of the beacon block proposer
+may not be available beyond the initial distribution via gossip.
+
+No more than `MAX_REQUEST_BLOCKS_EIP4844` may be requested at a time.
 
-`BeaconBlockAndBlobsSidecarByRoot` is primarily used to recover recent blocks and sidecars (e.g. when receiving a block or attestation whose parent is unknown).
+`BlobSidecarsByRoot` is primarily used to recover recent blocks and sidecars (e.g. when receiving a block or attestation whose parent is unknown).
 
 The response MUST consist of zero or more `response_chunk`.
-Each _successful_ `response_chunk` MUST contain a single `SignedBeaconBlockAndBlobsSidecar` payload.
+Each _successful_ `response_chunk` MUST contain a single `BlobSidecar` payload.
 
-Clients MUST support requesting blocks and sidecars since `minimum_request_epoch`, where `minimum_request_epoch = max(finalized_epoch, current_epoch - MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS, EIP4844_FORK_EPOCH)`. If any root in the request content references a block earlier than `minimum_request_epoch`, peers SHOULD respond with error code `3: ResourceUnavailable`.
+Clients MUST support requesting sidecars since `minimum_request_epoch`, where `minimum_request_epoch = max(finalized_epoch, current_epoch - MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS, EIP4844_FORK_EPOCH)`. If any root in the request content references a block earlier than `minimum_request_epoch`, peers MAY respond with error code `3: ResourceUnavailable` or not include the blob in the response.
 
-Clients MUST respond with at least one block and sidecar, if they have it.
+Clients MUST respond with at least one sidecar, if they have it.
```

Comment:
Do they need to respond in order that they are requested? that is if request is [1, 2, 3], and they respond [2, 1], is that malicious?

Reply:
we don't require that for blocks - also, it doesn't quite make sense since we need to compute the hash of the responses in order to know which blob ended up missing (ie if I receive `[X, X]` to a `[1, 2, 3]` request, even if they're in order I don't know which ones they are)